src/null-safety/index.md

```diff
@@ -86,10 +76,58 @@ Dart null safety support is based on the following three core design principles:
   fewer bugs, but smaller binaries and faster execution.
 
 
-## Enabling/disabling null safety {#enable-null-safety}
+## Dart 3 and null safety
+
+Dart 3---planned for a mid-2023 release---requires sound null safety. It
+will prevent code from running without it.
+
+Packages without null safety support will cause issues when resolving
+dependencies:
+
+```
+$ dart pub get
+
+Because mypackage doesn't support null safety, version solving failed.
+The lower bound of "sdk: '>=2.9.0 <3.0.0'" must be 2.12.0 or higher to enable null safety.
+```
+
+Libraries that opt out of null safety will cause analysis or compilation
+errors:
+
+```
+$ dart run bin/hi.dart
+
+../testpkg1/lib/testpkg1.dart:1:1: Error: Library doesn't support null safety.
```

Comment:
Same thing about package/library naming conventions (and `hi.dart` just looked strange to me, changed it to `test` but no big deal):
```suggestion
$ dart run bin/test.dart

../test_pkg/lib/test_pkg.dart:1:1: Error: Library doesn't support null safety.
```